pkg/volume/csi/csi_mounter.go

```diff
@@ -108,49 +110,87 @@ func (c *csiMountMgr) SetUpAt(dir string, fsGroup *int64) error {
 		return nil
 	}
 
-	csiSource, err := getCSISourceFromSpec(c.spec)
-	if err != nil {
-		klog.Error(log("mounter.SetupAt failed to get CSI persistent source: %v", err))
-		return err
-	}
-
 	csi := c.csiClient
 	ctx, cancel := context.WithTimeout(context.Background(), csiTimeout)
 	defer cancel()
 
-	// Check for STAGE_UNSTAGE_VOLUME set and populate deviceMountPath if so
-	deviceMountPath := ""
-	stageUnstageSet, err := csi.NodeSupportsStageUnstage(ctx)
+	volSrc, pvSrc, err := getSourceFromSpec(c.spec)
 	if err != nil {
-		klog.Error(log("mounter.SetUpAt failed to check for STAGE_UNSTAGE_VOLUME capabilty: %v", err))
+		klog.Error(log("mounter.SetupAt failed to get CSI persistent source: %v", err))
 		return err
 	}
 
-	if stageUnstageSet {
-		deviceMountPath, err = makeDeviceMountPath(c.plugin, c.spec)
-		if err != nil {
-			klog.Error(log("mounter.SetUpAt failed to make device mount path: %v", err))
-			return err
+	driverName := c.driverName
+	volumeHandle := c.volumeID
+	readOnly := c.readOnly
+	accessMode := api.ReadWriteOnce
+
+	var (
+		fsType             string
+		volAttribs         map[string]string
+		nodePublishSecrets map[string]string
+		publishContext     map[string]string
+		mountOptions       []string
+		deviceMountPath    string
+		secretRef          *api.SecretReference
+	)
+
+	switch {
+	case c.driverMode == ephemeralDriverMode && volSrc != nil && utilfeature.DefaultFeatureGate.Enabled(features.CSIInlineVolume):
+		if volSrc.FSType != nil {
+			fsType = *volSrc.FSType
 		}
-	}
-	// search for attachment by VolumeAttachment.Spec.Source.PersistentVolumeName
-	if c.publishContext == nil {
-		nodeName := string(c.plugin.host.GetNodeName())
-		c.publishContext, err = c.plugin.getPublishContext(c.k8s, c.volumeID, string(c.driverName), nodeName)
+
+		volAttribs = volSrc.VolumeAttributes
+
+		if volSrc.NodePublishSecretRef != nil {
+			secretName := volSrc.NodePublishSecretRef.Name
+			ns := c.pod.Namespace
+			secretRef = &api.SecretReference{Name: secretName, Namespace: ns}
+		}
+	case pvSrc != nil:
+		fsType = pvSrc.FSType
```

Comment:
do you want to check that driver mode is persistent here? (under the feature gate)

Reply:
done.